backend/db-setup/Dockerfile

```diff
@@ -1,8 +1,6 @@
-FROM postgres:14-alpine3.16
+FROM postgres:14-alpine3.17
 
 RUN apk add py3-pip postgresql-libs gcc musl-dev postgresql-dev make
-RUN pip3 install pgxnclient
-RUN pgxn install postgresql_anonymizer
```

Comment:
if we are removing this -- do we also want to clean up the DB anonymization feature work introduced here: https://github.com/CDCgov/prime-simplereport/pull/3887/files? Or are we keeping that in place for now? 🤔

Reply:
I appreciate your question. 

I would like to leave it in place for now; I still think it could be valuable in the long run, and this package works, just not with this version of Alpine as it is.

I'll create a new ticket (or find an old ticket) and see how it might be prioritized to either remove or fix this feature.

My priority in this PR is the Alpine upgrade, the ticket has been languishing, and I want to resolve a reported vulnerability with the older version of Alpine.